Source/WebCore/css/StyleProperties.cpp

```diff
@@ -952,6 +952,8 @@ String StyleProperties::getCommonValue(const StylePropertyShorthand& shorthand)
     String result;
     for (auto longhand : shorthand) {
         auto value = getPropertyCSSValue(longhand);
+        if (!value)
+            return String();
```

Comment:
If the bug is that https://searchfox.org/wubkat/rev/bfe2441d9e38a24d34f9bf9a8de6a19b07dd20a6/Source/WebCore/css/StyleProperties.cpp#1140-1143 is returning nullptr, even for non font-related properties (e.g. border), then the fix should be to check that the corresponding shorthand for the property in that codepath is font.

E.g.:

`shorthandForProperty(property.id()) == fontShorthand()`

I think this is an actual bug, `border: menu` successfully parses because `menu` is a color keyword, but at the same time, `menu` is a system font keyword, so the codepath above accidentally matches.

Reply:
Sorry `shorthandForProperty` is the wrong method to use.

I think changing the code linked in the previous comment to:
```
    // System fonts are represented as CSSPrimitiveValue for various font subproperties, but these must serialize as the empty string.
    // It might be better to implement this as a special CSSValue type instead of turning them into null here.
    if (property.shorthandID() == CSSPropertyFont && CSSPropertyParserHelpers::isSystemFontShorthand(valueID(value)))
        return nullptr;
```
should work?

Anyway the idea is that the check should only return true for font longhand properties, not any other, like `border`

Reply:
I agree that the change from
@nt1m looks like the correct fix.

Reply:
Sounds good, thanks for the feedback, I'll update my change to address that.